language-css fix exception in `insideURLFunctionInImportAtRuleNode`

romainmenke opened the pull request:
## Description

This fixes an exception when `node.groups` is empty in `insideURLFunctionInImportAtRuleNode`

## Checklist

<!-- Please ensure you’ve done all of these things (if applicable). -->
<!-- You can replace the `[ ]` with `[x]` to mark each task as done. -->

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

<!-- Please DO NOT remove the playground link -->

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

romainmenke commented on src/language-css/utils/index.js:
```diff
@@ -95,7 +95,7 @@ function insideAtRuleNode(path, atRuleNameOrAtRuleNames) {
   const { node } = path;
   return (
-    node.groups[0].value === "url" &&
+    node.groups?.[0].value === "url" &&
```
I don't know exactly which source code triggers the exception.
I encountered this condition when running prettier through Stylelint with their VSCode extension during or shortly after a switch between branches.

I mimicked the same code as found in `isKeyValuePairInParenGroupNode`.
Logically this change should prevent exceptions and should still work as intended.

sosukesuzuki commented:
We don't know the problem fixed by this change. So can you show the exception in https://prettier.io/playground/ ?

And at least please add tests.

romainmenke commented:
@sosukesuzuki I've found a way to trigger the exception in the playground and added a link to it in the original comment.

I've also added tests and a more complete fix for the issue.

sosukesuzuki commented:
Your tests are failed with `FULL_TEST=1`. Please see https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md#deeper-testing and fix it.

romainmenke commented:
@sosukesuzuki I don't think these tests can pass.

The full test suite checks that a second run doesn't have any further effect.
But in this case it does.

The first pass :

```diff
- @import ,;
+ @import ;
```

second pass :

```diff
- @import ;
+ @import;
```

Is there a way to cover this with a unit test and not with a snapshot test?
Something that tests that Prettier doesn't throw when encountering such malformed CSS?

fisker commented:
We can ignore second format https://github.com/prettier/prettier/blob/0661f78a2603b52f2382dc3c7639014e16999e5a/tests/config/format-test.js#L22, or try to print a `,` there.

romainmenke commented:
@fisker Do you have an example of a different case where `,` is printed?

I don't know the Prettier code base or how it works internally, so I am not even sure why the `,` goes away in the first place :)

I don't think the entire css `at-import` tests should be marked as unstable, that will make it harder to catch other issues and this is more intended as a code quality patch.

fisker commented:
put  the input to `test.css`, use `yarn debug test.css` to find how it's printed.

romainmenke commented:
I meant another location in the Prettier codebase where a `,` is added to the output.
If I can see how this is done for other cases then maybe I can figure our how to do it here.

fisker commented:
I don't believe there is an existing case like this.

romainmenke commented:
It's also fine for me to leave this as is :)
I thought it was an obvious bug fix and this isn't blocking for me in anyway.
